### PR TITLE
fix parisng of tasks env vars in go

### LINKED ISSUE #5699: .gitpod.yml tasks not running
### Bug description

Literally just the title. The workspace is created but the tasks in .gitpod.yml aren't run

### Steps to reproduce

Just create a new workspace.

### Expected behavior

_No response_

### Example repository

github.com/kronifer/sir-lancebot/tree/gitpod

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

Our gitpod protocol schema allows any as a type for env vars of tasks. This PR adjusts typings to respect it in the code.
We considered also to make the scheme more strict, but it will be a breaking change, and there is no apparent downside of any value given that clients expect that the value be serialized/deserialized as JSON. 

## Related Issue(s)

fix #5699

## How to test

- Start a workspace on the repo which is using not string env var for tasks: https://akosyakov-gitpod-yml-tasks-not-5699.staging.gitpod-dev.com/#https://github.com/akosyakov/sir-lancebot/blob/eca8bcebf7c47de7fa661009df583fe79b95334e/.gitpod.yml#L5
- Check that the task terminal is running and env var is set: `printenv PIP_USER` should be `false`.

## Release Notes

```release-note
Fix parsing of tasks' env vars.
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#5699*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: c00fea839c25beb1a97227d65278ca2fb87f90e4</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5719#pullrequestreview-756961599" title="LGTM">jeanp413</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#5699*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5719#issuecomment-921615262" title="Approved">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5719#pullrequestreview-756961599" title="LGTM">jeanp413</a>*

Associated issue: *#5699*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [JanKoehnlein]
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
